lua/entities/gmod_wire_hologram.lua

```diff
@@ -100,28 +100,30 @@ if CLIENT then
 		end
 	end
 
-	function ENT:FinishClipping()
-		if next(self.clips) then
-			for _, clip in pairs(self.clips) do
+	function ENT:FinishClipping(selfTbl)
+		if next(selfTbl.clips) then
+			for _, clip in pairs(selfTbl.clips) do
 				if clip.enabled and clip.normal and clip.origin then -- same logic as in SetupClipping
 					render.PopCustomClipPlane()
 				end
 			end
 
-			render.EnableClipping(self.oldClipState)
+			render.EnableClipping(selfTbl.oldClipState)
 		end
 	end
 
 	function ENT:Draw()
-		if self.blocked or self.notvisible then return end
+		local selfTbl = self:GetTable()
+		if selfTbl.blocked or selfTbl.notvisible then return end
 
-		if self:GetColor().a ~= 255 then
-			self.RenderGroup = RENDERGROUP_BOTH
+		local _, _, _, alpha = self:GetColor4Part()
+		if alpha ~= 255 then
+			selfTbl.RenderGroup = RENDERGROUP_BOTH
 		else
-			self.RenderGroup = RENDERGROUP_OPAQUE
+			selfTbl.RenderGroup = RENDERGROUP_OPAQUE
 		end
 
-		self:SetupClipping()
+		self:SetupClipping(selfTbl)
```

Comment:
any idea if the 'self' function call here invokes __index?

Reply:
It will but it will exit relatively soon https://github.com/Facepunch/garrysmod/blob/98bdd1ea744d3b18a9c3ae402b0f72dadd8d1a21/garrysmod/lua/includes/extensions/entity.lua#L18-L24

I could localize the methods too, would some extra perf

Reply:
These methods don't exist in 'meta' so it'll actually need to GetTable I think. Would be a good idea to make those local functions.